packages/hint-compat-api/src/types.ts

```diff
@@ -10,20 +10,22 @@ export type BrowserSupportCollection = {
 
 export type FeatureStrategy<T extends ChildNode> = {
     check: (node: T | ChildNode) => boolean;
-    testFeature: (node: T, data: MDNTreeFilteredByBrowsers, location?: ProblemLocation) => void;
+    testFeature: (node: T, location?: ProblemLocation) => void;
 };
 
-export type TestFeatureFunction = (feature: FeatureInfo, supportBlock: SupportBlock) => void;
+export type TestFeatureFunction = (feature: FeatureInfo, collection: CompatStatement | undefined) => void;
 
 export type BrowserVersions = {
     [key: string]: string[];
 };
 
+// eslint-disable: no-use-before-define, typescript/no-use-before-define
```

Comment:
Why do you need this?

Anyways, enable the rule as soon it is not necessary.

Reply:
As you can see, I’m using the same type in a child property. The linting was warning me because I was using the type before defining it. I guess the rule still makes sense in other scenario, so I just disabled it in this context.

Reply:
That is weird, you should be able to use the same type you are defining as the type of one of its property without any warning...

Reply:
For some reason, now eslint is ok without disabling the rule. I don't understand why I was getting the warning 😩 . I just removed this line 👍